util/threadpool_imp.cc

```diff
@@ -90,7 +90,8 @@ struct ThreadPoolImpl::Impl {
   }
 
   bool IsExcessiveThread(size_t thread_id) const {
-    return static_cast<int>(thread_id) >= total_threads_limit_;
+    return static_cast<int>(thread_id) + reserved_threads_ >=
+           total_threads_limit_;
```

Comment:
This seems inconsistent with `HasExcessiveThread()`. Personally I'd not terminate/recreate thread-pool threads for reservations.

Reply:
I see your concern. Will change it.

Reply:
Oh good point. I guess we could resolve the inconsistency either by changing the name `IsExcessiveThread()` to something else like `IsBlockedThread()`, or by adding back the `reserved_threads_` into `HasExcessiveThread()`. Up to you.

Reply:
I am not quite following the why the new `IsBlockedThread()` implementation is correct in preventing extra threads being woken up as a result of not considering reserved threads. See (4) in my review summary https://github.com/facebook/rocksdb/pull/10278#issuecomment-1175764224 for more.